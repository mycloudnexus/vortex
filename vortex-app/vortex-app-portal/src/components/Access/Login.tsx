/* eslint-disable @typescript-eslint/no-unused-vars */

import React, { useCallback, useEffect } from 'react'
import { useNavigate, useParams } from 'react-router-dom'
import { useAuth0 } from '@auth0/auth0-react'
import { Spin } from 'antd'
import { ENV } from '@/constant'
import { storeOrg, storeToken } from '@/utils/helpers/token'
import { useAppStore } from '@/stores/app.store'
import type { AuthUser } from '@/stores/app.store'

const Login = () => {
  const { getAccessTokenSilently, isLoading, isAuthenticated, loginWithRedirect, user } = useAuth0()
  const navigate = useNavigate()
  const { organization = ENV.AUTH0_MGMT_ORG_ID } = useParams()
  const { setCurrentAuth0User } = useAppStore()

  const saveToken = useCallback(async () => {
    const res = await getAccessTokenSilently()
    storeToken(res)
  }, [getAccessTokenSilently])

  useEffect(() => {
    storeOrg(organization!)
  }, [organization])

  useEffect(() => {
    if (!isAuthenticated && !isLoading) {
      loginWithRedirect({
        authorizationParams: {
          organization,
          redirect_uri: window.location.origin
        }
      })
    }
    if (isAuthenticated) {
      saveToken()
    }
  }, [saveToken, isAuthenticated, isLoading, loginWithRedirect, organization])

  useEffect(() => {
    if (!user) return
    setCurrentAuth0User(user as AuthUser)
    navigate('/')
  }, [user, setCurrentAuth0User])

  if (isLoading || !isAuthenticated) {
    return <Spin />
  }

  return <></>
}

export default Login
