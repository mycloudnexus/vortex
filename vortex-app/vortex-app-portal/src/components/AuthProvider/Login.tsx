/* eslint-disable @typescript-eslint/no-unused-vars */

import React, { useCallback, useEffect, ReactNode } from 'react'
import { useNavigate, useParams } from 'react-router-dom'
import { useAuth0 } from '@auth0/auth0-react'
import { Spin } from 'antd'
import { useAppStore } from '@/stores/app.store'
import type { AuthUser } from '@/stores/app.store'

// 'org_doRtBm1mKsGlUw3W' //usename password
//  'org_rKgjdXyGVS37FUuo'//consoleconnect

const Login = () => {
  // const organization = window.localStorage.getItem('org') || 'org_rKgjdXyGVS37FUuo'
  const { getAccessTokenSilently, isLoading, isAuthenticated, loginWithRedirect, user } = useAuth0()
  const navigate = useNavigate()
  const { organization = 'org_rKgjdXyGVS37FUuo' } = useParams()
  console.log('-organizationorganization', organization)
  const { setCurrentUser } = useAppStore()

  const saveToken = useCallback(async () => {
    const res = await getAccessTokenSilently()
    window.localStorage.setItem('token', res)
  }, [getAccessTokenSilently])

  useEffect(() => {
    if (!isAuthenticated && !isLoading) {
      loginWithRedirect({
        authorizationParams: {
          organization,
          redirect_uri: window.location.origin
        }
      })
    }
    if (isAuthenticated) {
      saveToken()
    }
  }, [saveToken, isAuthenticated, isLoading, loginWithRedirect, organization])

  useEffect(() => {
    if (!user) return
    setCurrentUser(user as AuthUser)
    navigate('/')
  }, [user, setCurrentUser])

  if (isLoading || !isAuthenticated) {
    return <Spin />
  }

  return <></>
}

export default Login
