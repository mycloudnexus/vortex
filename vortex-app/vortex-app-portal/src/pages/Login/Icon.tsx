export const Logo = () => (
  <svg width='37' height='36' viewBox='0 0 37 36' fill='none' xmlns='http://www.w3.org/2000/svg'>
    <g clipPath='url(#clip0_2848_34986)'>
      <path
        d='M13.625 2.24829C16.625 1.37329 23.975 0.823293 29.375 5.62329C36.125 11.6233 35 20.2484 34.25 22.8734'
        stroke='url(#paint0_linear_2848_34986)'
        strokeWidth='2.6'
        strokeLinecap='round'
      />
      <path
        d='M10.8804 13.4768L18.4712 9.06871L26.0404 13.4394L26.0188 22.2183L18.428 26.6264L10.8588 22.2557L10.8804 13.4768Z'
        fill='#2962FF'
      />
      <path
        d='M4.12288 8.28524L4.13171 4.69692L7.23443 2.89511L10.3216 4.67775L10.3128 8.26608L7.21006 10.0679L4.12288 8.28524Z'
        stroke='#2962FF'
        strokeWidth='1.55594'
      />
      <path
        d='M26.5951 27.6517L30.8592 25.1755L35.1111 27.6306L35.0989 32.562L30.8349 35.0383L26.583 32.5831L26.5951 27.6517Z'
        fill='#9156E4'
      />
      <path
        d='M23.1992 34.1228C20.1992 34.9978 12.8492 35.5478 7.44922 30.7478C0.69922 24.7478 1.82422 16.1227 2.57422 13.4977'
        stroke='url(#paint1_linear_2848_34986)'
        strokeWidth='2.6'
        strokeLinecap='round'
      />
    </g>
    <defs>
      <linearGradient
        id='paint0_linear_2848_34986'
        x1='24.2233'
        y1='1.66406'
        x2='24.2233'
        y2='22.8734'
        gradientUnits='userSpaceOnUse'
      >
        <stop stopColor='#2962FF' />
        <stop offset='1' stopColor='#9156E4' />
      </linearGradient>
      <linearGradient
        id='paint1_linear_2848_34986'
        x1='12.6009'
        y1='34.707'
        x2='12.6009'
        y2='13.4977'
        gradientUnits='userSpaceOnUse'
      >
        <stop stopColor='#2962FF' />
        <stop offset='1' stopColor='#9156E4' />
      </linearGradient>
      <clipPath id='clip0_2848_34986'>
        <rect width='36' height='36' fill='white' transform='translate(0.5)' />
      </clipPath>
    </defs>
  </svg>
)
