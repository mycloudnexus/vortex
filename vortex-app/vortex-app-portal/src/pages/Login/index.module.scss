.container {
  display: flex;
  width: 100vw;
  height: 100vh;
  .bgContainer {
    width: 69%;
    height: 100%;
    background: url('./bg.svg') no-repeat;
    background-size: cover;
  }
  .flexContainer {
    flex: 1;
    img {
      width: 200px;
    }
    h1 {
      color: #000;
      text-align: center;
      font-size: 18px;
      font-style: normal;
      font-weight: 600;
      line-height: 150%;
      margin: 0;
    }
    p {
      margin: 0;
      margin-bottom: 24px;
      color: #000;
      font-size: 14px;
      font-style: normal;
      font-weight: 400;
      line-height: 150%;
      /* 21px */
    }
    :global(.ant-btn) {
      width: calc(100% - 100px);
    }
  }
}
