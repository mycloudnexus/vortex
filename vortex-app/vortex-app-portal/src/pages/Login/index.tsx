import { useEffect, useMemo } from 'react'
import { useNavigate, useParams, useLocation } from 'react-router-dom'
import { useAuth0 } from '@auth0/auth0-react'
import { Flex, Button, Skeleton } from 'antd'
import { storeOrg } from '@/utils/helpers/token'
import { useAppStore } from '@/stores/app.store'
import { Logo } from './Icon'
import type { AuthUser } from '@/stores/type'
import * as styles from './index.module.scss'

const Login = () => {
  const { loginWithRedirect, user, error } = useAuth0()
  const navigate = useNavigate()
  const { organization } = useParams()
  const { setCurrentAuth0User, setuserType } = useAppStore()
  const { search } = useLocation()
  const searchParams = new URLSearchParams(search?.slice(1))

  useEffect(() => {
    if (organization) {
      storeOrg(organization)
      setuserType('customer')
    } else {
      setuserType('reseller')
    }
  }, [organization])

  const authorizationParams = useMemo(() => {
    const url = window.location.origin
    const obj = { redirect_uri: organization ? url : `${url}/login` }
    return organization ? { ...obj, organization } : obj
  }, [organization])

  useEffect(() => {
    if (!user) return
    setCurrentAuth0User(user as AuthUser)
    navigate('/')
  }, [user, setCurrentAuth0User])

  if (error) {
    return <div>{error.message}</div>
  }

  if (searchParams.get('code')) {
    return <Skeleton />
  }

  return (
    <div className={styles.container}>
      <div className={styles.bgContainer}></div>
      <Flex vertical align='center' justify='center' className={styles.flexContainer}>
        <Flex align='center'>
          <Logo />
          <b style={{ marginLeft: 4, fontSize: 20, fontWeight: 700, color: '#292929' }}>goPartnerConnect</b>
        </Flex>
        <h1 style={{ marginTop: 24 }}>ACCOUNT LOGIN</h1>
        <p>Log in to goPartnerConnect.</p>
        <Button
          type='primary'
          onClick={() => {
            loginWithRedirect({ authorizationParams })
          }}
        >
          Log in
        </Button>
      </Flex>
    </div>
  )
}

export default Login
