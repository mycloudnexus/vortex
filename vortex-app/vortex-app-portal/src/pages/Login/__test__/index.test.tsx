import { render, fireEvent } from '@testing-library/react'
import { useAuth0 } from '@auth0/auth0-react'
import Login from '..'

jest.mock('@auth0/auth0-react')

const mockedUsedNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
  useParams: jest.fn().mockReturnValue({})
}))

const fn = useAuth0 as any
describe('Login page', () => {
  it('login as a reseller', () => {
    const url = 'https://www.test.com'
    Object.defineProperty(window, 'location', {
      value: {
        origin: url
      },
      writable: true
    })
    const loginWithRedirect = jest.fn()
    fn.mockReturnValue({
      loginWithRedirect,
      isAuthenticated: false,
      error: false
    })
    const { getByText } = render(<Login />)
    const loginBtn = getByText('Log in')
    expect(loginBtn).toBeTruthy()
    fireEvent.click(loginBtn)

    expect(loginBtn).toBeTruthy()
    expect(loginWithRedirect).toHaveBeenCalledWith({
      authorizationParams: {
        redirect_uri: 'https://www.test.com/login'
      }
    })
  })
  it('login as a customer', () => {
    jest.mock('react-router-dom', () => ({
      useParams: jest.fn().mockReturnValue({ organization: 'orgid123' })
    }))
    const origin = 'https://www.test.com'
    const url = origin + '/orgid123/login'
    Object.defineProperty(window, 'location', {
      value: {
        href: url,
        origin
      },
      writable: true
    })
    const loginWithRedirect = jest.fn()
    fn.mockReturnValue({
      loginWithRedirect,
      isAuthenticated: false,
      error: false
    })
    const { getByText } = render(<Login />)
    const loginBtn = getByText('Log in')
    expect(loginBtn).toBeTruthy()
    fireEvent.click(loginBtn)

    expect(loginBtn).toBeTruthy()
    expect(loginWithRedirect).toHaveBeenCalledWith({
      authorizationParams: {
        redirect_uri: origin + '/login'
      }
    })
  })

  it('should redirect index page if have user', () => {
    fn.mockReturnValue({ isAuthenticated: true, user: {}, getAccessTokenSilently: jest.fn() } as any)
    render(<Login />)
    expect(mockedUsedNavigate).toHaveBeenCalledTimes(1)
    expect(mockedUsedNavigate).toHaveBeenCalledWith(`/`)
  })
})
