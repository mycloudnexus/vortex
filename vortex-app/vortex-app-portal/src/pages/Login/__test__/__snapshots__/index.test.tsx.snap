// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should show Skeleton if url includes code search params  1`] = `
<div
  class="ant-skeleton css-dev-only-do-not-override-49qm"
>
  <div
    class="ant-skeleton-content"
  >
    <h3
      class="ant-skeleton-title"
      style="width: 38%;"
    />
    <ul
      class="ant-skeleton-paragraph"
    >
      <li />
      <li />
      <li
        style="width: 61%;"
      />
    </ul>
  </div>
</div>
`;
